### PR TITLE
Allow unified user definitions parser to restart

### PR DESCRIPTION
**Pull request by rick-nguyen**
Similar to previous PR: https://github.com/microsoft/Power-Fx/pull/1153

Restart no longer works with named formulas under the unified parser. This PR fixes that and allows UDFs to continue processing as well if the parser encounters an error.

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs [hunk 1]
```diff
@@ -135,7 +136,14 @@ private bool ParseUDFArgs(out HashSet<UDFArg> args)
                 if (TokEat(TokKind.Colon, addError: false) == null)
                 {
                     CreateError(_curs.TokCur, TexlStrings.ErrUDF_MissingParamType);
-                    break;
+
+                    // If the result was an error, keep moving cursor until end of named formula expression
```
Comment (lesaltzm) [inline src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs:140]: nit: end of expression (not just named formula)?

File: src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs [hunk 2]
```diff
@@ -135,7 +136,14 @@ private bool ParseUDFArgs(out HashSet<UDFArg> args)
                 if (TokEat(TokKind.Colon, addError: false) == null)
                 {
                     CreateError(_curs.TokCur, TexlStrings.ErrUDF_MissingParamType);
-                    break;
+
+                    // If the result was an error, keep moving cursor until end of named formula expression
+                    while (_curs.TidCur != TokKind.Semicolon && _curs.TidCur != TokKind.Eof)
```
Comment (lesaltzm) [inline src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs:141]: If it's missing a param type, could we try to move to the `)` or `;`? maybe the body is still valid? not for this pr, but something to consider, udfs have some other syntax elements that can be used.

**Review by lesaltzm — APPROVED** [review]

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by Shpakh-MSFT — APPROVED** [review]
:shipit:

File: src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs [hunk 3]
```diff
@@ -144,11 +152,24 @@ private bool ParseUDFArgs(out HashSet<UDFArg> args)
                 if (varType == null)
                 {
                     CreateError(_curs.TokCur, TexlStrings.ErrUDF_MissingParamType);
+
+                    // If the result was an error, keep moving cursor until end of expression
+                    while (_curs.TidCur != TokKind.Semicolon && _curs.TidCur != TokKind.Eof)
+                    {
+                        _curs.TokMove();
+                    }
```
Comment (Shpakh-MSFT) [inline src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs:160]: This could be a helper as it is repeated multiple times.

**Pull request merged**